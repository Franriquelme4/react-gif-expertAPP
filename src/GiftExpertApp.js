import { Fragment, useState } from "react";
import React from 'react';
import AddCategory from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

const GiftExpertApp = () => {
 const [categories,setCategories] = useState([]);
  return (
    <Fragment>
      <h2>GiftExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      <br/>
      <button onClick={()=>setCategories([])}>Reset</button>
      <hr />
        <ol>
          {
            categories.map(category => (<GifGrid 
                                            key={category}
                                            category={category}
                                            />))
          }
        </ol>
    </Fragment>
  );
};

export default GiftExpertApp;
