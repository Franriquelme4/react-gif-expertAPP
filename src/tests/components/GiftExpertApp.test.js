
import '@testing-library/jest-dom';
import React from 'react';
//import { render } from "@testing-library/react";
import { shallow } from "enzyme";
import GiftExpertApp from '../../GiftExpertApp';
describe('SE prueba la carga de los componentes',()=>{

    test('Se prueba la carga de los componentes', () => { 
        const wrapper =  shallow(<GiftExpertApp/>);

        expect(wrapper).toMatchSnapshot();
        


     })



})