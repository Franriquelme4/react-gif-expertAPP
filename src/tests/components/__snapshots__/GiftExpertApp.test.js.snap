// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SE prueba la carga de los componentes Se prueba la carga de los componentes 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <br />
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <hr />
  <ol>
    <GifGrid
      category="cats"
      key="cats"
    />
  </ol>
</Fragment>
`;
